src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -48,6 +48,13 @@ import scala.collection.mutable.ListBuffer
 trait SamRecordClipper {
   import SamRecordClipper._
 
+  /**
+    * Should extended attributes that are the same length as the bases and qualities
+    * be automatically clipped in the same way as the bases and qualities.
+    * @return
+    */
+  def autoClipAttributes: Boolean
```

Comment:
I am now thinking that `ClippingMode` should also be a `def clippingMode: ClippingMode` just like this, since when are you going to want different clipping modes?  Or am I wrong about that?